drivers/spi/spi_esp32_spim.c

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_esp32_spi
+
+/* Include esp-idf headers first to avoid redefining BIT() macro */
+#include <hal/spi_hal.h>
+#include <esp_attr.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(esp32_spi);
+
+#include <soc.h>
+#include <drivers/spi.h>
+#include <drivers/gpio/gpio_esp32.h>
+#include <drivers/clock_control.h>
+#include "spi_context.h"
+#include "spi_esp32_spim.h"
+
+#define DEV_CFG(dev) (struct spi_esp32_config *const)(dev->config)
+#define DEV_DATA(dev) (struct spi_esp32_data *const)(dev->data)
+
+K_TIMER_DEFINE(spi_done_timer, NULL, NULL);
+
+static inline void spi_esp32_complete(struct spi_esp32_data *data,
+				      spi_dev_t *spi, int status)
+{
+	spi_context_cs_control(&data->ctx, false);
+}
+
+static int IRAM_ATTR spi_esp32_transfer(const struct device *dev)
```

Comment:
Would be good to write some comment in commit message why `IRAM_ATTR` is used here (e.g. some hardware limitation / requirement). Just so there is some hint for people refactoring this code in future.

Reply:
Of course, I can add this info on commit message. However, this attribute is and will be used in lots of other places as well. 
Anyway, for more information:  
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/general-notes.html#application-memory-layout